node/src/services/handler.rs

```diff
@@ -0,0 +1,175 @@
+use sp_trie::TrieLayout;
+use std::sync::Arc;
+use tokio::sync::RwLock;
+
+use file_manager::{in_memory::InMemoryFileStorage, traits::FileStorage};
+use forest_manager::{
+    in_memory::InMemoryForestStorage, rocksdb::RocksDBForestStorage, traits::ForestStorage,
+};
+use storage_hub_infra::{
+    actor::{ActorHandle, TaskSpawner},
+    event_bus::EventHandler,
+};
+
+use crate::tasks::{bsp_volunteer_mock::BspVolunteerMockTask, user_sends_file::UserSendsFileTask};
+
+use super::{blockchain::handler::BlockchainService, file_transfer::FileTransferService};
+
+/// Wrapper trait encompassing all the necessary implementations which [`StorageHubHandler`] requires.
+pub trait StorageHubHandlerConfig: StorageHubHandlerInitializer + Send + 'static {
+    /// Type which implements [`TrieLayout`].
+    ///
+    /// This is primarily used for constructing
+    type TrieLayout: TrieLayout;
+    /// Type which implements [`FileStorage`].
+    ///
+    /// This layer stores all files (chunked).
+    type FileStorage: FileStorage<Self::TrieLayout> + Send + Sync;
+    /// Type which implements [`ForestStorage`].
+    ///
+    /// This layer tracks all of the files stored in [`FileStorage`](StorageHubHandlerConfig::FileStorage).
+    type ForestStorage: ForestStorage<Self::TrieLayout> + Send + Sync;
+}
+
+pub struct InMemoryStorageHubConfig<T: TrieLayout>(std::marker::PhantomData<T>);
+
+impl<T: TrieLayout + Send + 'static> StorageHubHandlerConfig for InMemoryStorageHubConfig<T>
+where
+    <<T as TrieLayout>::Hash as sp_core::Hasher>::Out: TryFrom<[u8; 32]>,
```

Comment:
If we are binding the hash size to 32 bytes, why not using `H256` directly for `Key` and `ForestProof`?